feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -259,6 +323,26 @@ class PreviewViewModel @AssistedInject constructor(
         }
     }
 
+    fun showToastForDisabledHdrToggle() {
+        val disabledReason =
+            (previewUiState.value as? PreviewUiState.Ready)?.captureModeToggleUiState
+                ?.disabledReason ?: throw RuntimeException(
+                "showing Hdr toggle disabled toast without disabled reason."
+            )
```

Comment:
Now you can have the `disabledReason` passed in as an argument to `showToastForDisabledHdrToggle` rather than polling the UiState.

That way we can avoid the `as?` (or any of the `?` operators) returning null in case the previewUiState has changed between tapping the disabled capture toggle and calling this method.